### ISSUE TITLE
Off-by-one diagnostic

### ISSUE THREAD

**Issue by mattmassicotte** [castwide/solargraph#595]
After a little digging, I've discovered that the following document will produce a diagnostic with a position that is 1 character past the end of the last line. Something with the line being empty matters here. If it is a single space, things are ok.

```ruby
require 'a thing'

class Example
  # @return [String]
  def stringify
    "example".
  end
end

Example.new.stringify(1, 2)
Example.new.stringify(1, 2)

```

The diagnostic is:

```
unexpected token $end
(Using Ruby 2.5 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)
```

rubocop.yml
```yaml
Style/StringLiterals:
  EnforcedStyle: double_quotes

```

**Comment by castwide** [comment]
That diagnostic gets reported because of the trailing period in `"example".`. Solargraph can ignore those kinds of syntax errors under certain circumstances, but RuboCop still detects it.

**Comment by mattmassicotte** [comment]
Just to be really clear, the **diagnostic** itself is expected. The range it returns is out of bounds.

**Comment by castwide** [comment]
Ah, gotcha. RuboCop uses 1-based lines and columns, and there's a branch in the reporter that doesn't account for it. I'll fix it in a patch.

**Comment by castwide** [comment]
Released in v0.47.2.
